### PR TITLE
fix(portrait): rewrite types and add invariant for gravatar and src p…

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Only one type interface will exist on the portrait component, named `PortraitProps` which will include all props from `PortraitBaseProps` and `PortraitWithSrc/PortraitWithGravatar`, an invariant has also been added to throw a better formatted and more informative error message to consumers with the `src` and `gravatar` props are used in conjunction.

### Current behaviour

Currently, the interfaces `PortraitWithGravatar` and `PortraitWithSrc` are used to throw a type error when both the `src` and `gravatar` props are passed at the same time. This results in a type error which is not very descriptive or helpful, and could be handled better.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/portrait/portrait.component.tsx [hunk 1]
```diff
@@ -89,6 +80,12 @@ export const Portrait = ({
   const { roundedCornersOptOut } = useContext(RoundedCornersOptOutContext);
   const defaultShape = roundedCornersOptOut ? "square" : "circle";
 
+  invariant(
+    !(src && gravatar),
+    "The `src` pop cannot be used in conjunction with the `gravatar` prop." +
```
Comment (DipperTheDan) [inline src/components/portrait/portrait.component.tsx:85]:
```suggestion
    "The `src` prop cannot be used in conjunction with the `gravatar` prop." +
```

File: src/components/portrait/portrait.spec.tsx [hunk 2]
```diff
@@ -380,6 +380,37 @@ describe("PortraitComponent", () => {
     expect(wrapper.find(Tooltip).exists()).toBeTruthy();
   });
 
+  describe("invariant", () => {
+    it("validates an invariant is thrown if `src` and `gravatar` are set at the same time", () => {
+      const consoleSpy = jest.spyOn(console, "error");
+      consoleSpy.mockImplementation(() => {});
+
+      expect(() => shallow(<Portrait src="foo" gravatar="baz" />)).toThrow(
+        "The `src` pop cannot be used in conjunction with the `gravatar` prop. Please use one or the other."
```
Comment (DipperTheDan) [inline src/components/portrait/portrait.spec.tsx:389]:
```suggestion
        "The `src` prop cannot be used in conjunction with the `gravatar` prop. Please use one or the other."
```

**Review by edleeks87 — COMMENTED** [review]
nothing else from me

**Review by DipperTheDan — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**